### PR TITLE
Issue #11588: Resolve SimplifiableIfStatement inspection violations

### LINKED ISSUE #11588: Resolve SimplifiableIfStatement inspection violations
Taken from https://github.com/checkstyle/checkstyle/issues/11566

see commit: https://github.com/checkstyle/checkstyle/commit/009cffaae0f1bd79f5c0084f5354f45ed391e346

we need to revert suppression of SimplifiableIfStatement or explain why they should be permanent.

### PR DESCRIPTION
**Pull request by nick-mancuso**
Closes #11588.

Diff Regression config: https://gist.githubusercontent.com/nick-mancuso/858f3ef5e63e12540b71a3e514acdf52/raw/b9e3d229bb0e226cfb11b98e5e3c26895c26faa7/config.xml

### CODE REVIEW

**Comment by nick-mancuso** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java [hunk 1]
```diff
@@ -728,25 +733,6 @@ private void customLeaveToken(DetailAST ast, Deque<VariableDesc> variablesStack)
...
-    /**
-     * Whether an ident with parent node of type {@link TokenTypes#METHOD_REF}
-     * should be checked or not.
-     *
-     * @param identAst identAst
-     * @return true if an ident with parent node of type {@link TokenTypes#METHOD_REF}
-     *         should be checked or if the parent type is not {@link TokenTypes#METHOD_REF}
-     * @noinspection SimplifiableIfStatement
-     */
-    public static boolean shouldCheckIdentWithMethodRefParent(DetailAST identAst) {
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java:740]: Name of method was misleading; this method checked ASTs with many different parents such as type declarations and object initialization. I have removed the method entirely and just used two booleans in `visitIdentToken` above.

**Comment by nick-mancuso** [comment]
Github, generate report

**Comment by nick-mancuso** [comment]
Github, generate report

**Comment by romani** [comment]
False positives in report.

**Comment by nick-mancuso** [comment]
Github, generate report

**Review by pbludov — APPROVED** [review]

**Comment by nick-mancuso** [comment]
Github, generate report

**Review by romani — APPROVED** [review]

**Pull request merged**